wgengine/userspace.go

```diff
@@ -1362,3 +1363,18 @@ func (p closeOnErrorPool) closeAllIfError(errp *error) {
 		}
 	}
 }
+
+// fwdDNSlinkSelector is userspaceEngine's resolver.ForwardLinkSelector, to pick
+// which network interface to send DNS queries out of.
+type fwdDNSlinkSelector struct {
+	ue      *userspaceEngine
+	tunName string
+}
+
+func (ls fwdDNSlinkSelector) PickLink(ip netaddr.IP) (linkName string) {
+	n, err := ls.ue.peerForIP(ip)
```

Comment:
This is likely wrong for exit nodes. I don't think peerForIP handles that sufficiently.